### ISSUE TITLE
Deleted apps can still be accessed

### ISSUE THREAD

**Issue by Morganamilo** [ably/terraform-provider-ably#148]
### Problem
After an app is deleted it disappears from `/apps`, but `GET /apps/<appId>/keys` still returns its keys. Either the app is not really gone or the keys endpoint does not check that its app still exists. Both readings are a problem: the first means deletion is incomplete, the second means an API returns credentials belonging to something that no longer exists.

Priority is raised from Unprioritized because this concerns credentials surviving a deletion, not a cosmetic inconsistency.

### Context
This is a Control API behaviour rather than a provider bug. The provider deletes an app through the Control API and removes it from state; it has no say in what the keys endpoint returns afterwards.

The report as written is thin: it does not say whether the keys returned are still usable for authentication, which is the difference between an API tidiness bug and something that needs handling urgently. That is the first thing to determine.

Nonprod reproduction is currently awkward because there is no Control API for nonprod clusters, which is INF-6921.

This is a Unito mirror of a GitHub issue on `ably/terraform-provider-ably` and predates the current provider.

### Acceptance criteria
* Reproduced against a current environment, with the exact requests and responses recorded, or closed as no longer reproducible
* Established whether keys returned for a deleted app still authenticate; if they do, that is raised separately and immediately rather than waiting on this ticket
* The behaviour is either fixed in the Control API or tracked against a Control API ticket with this one closed as a mirror
* If deletion is asynchronous by design, that is documented, along with how long the window is

### Scope
* **In scope:** establishing what actually happens today and getting it in front of whoever owns the Control API
* **Out of scope:** provider changes, since the provider is not the cause, and any change to how apps are deleted

### Pointers
* INF-6921 - the missing nonprod Control API that makes this hard to reproduce safely
* `internal/provider/resource_ably_app.go` - how the provider deletes an app
* The website repository, which is where the Control API lives
* GitHub issue on `ably/terraform-provider-ably`

┆Issue is synchronized with this [Jira Bug](https://ably.atlassian.net/browse/INF-5479) by [Unito](https://www.unito.io)
